e2e/src/app.js

```diff
@@ -1,38 +1,104 @@
+        msgs.map((ms) => ({
+            user: ms.userId,
+            text: ms.text
+        })).forEach((mc) => {
+            addMessageText('messages', 'p', mc);
+        });
     });
+};
 
-    const _messageFactory = messageFactory('random-channel', 'pepe');
-    const sendbutton = document.getElementById('sendbutton');
-    sendbutton.addEventListener('click', (evt) => {
-        evt.stopPropagation();
-        const textElement = document.getElementById('message');
-        const message = _messageFactory.compose(textElement.value)
-        clearElement(textElement);
-        socket.emit('message', message); 
-        // TODO: Move this code to message handler.
-        const messageElement = document.createElement('p');
-        messageElement.innerText = `${message.user}: ${message.text}`;
-        document.getElementById('messages').appendChild(messageElement);
+const handleSelectedRoom = () => {
+    // NOTE: Bear on mind to disconnect from socket channel.
+    document.getElementById('rooms')
+        .addEventListener('click', (evt) => {
+            evt.stopPropagation();
+            changeSelectedRoom('selectedroom', evt.target.innerHTML);
+        });
+};
+
+const handleEnrolment = (callback) => {
+    // NOTE: Check selected room.
+    // NOTE: Check input user.
+    document.getElementById('enrolmentsubmit')
+        .addEventListener('click', (evt) => {
+            evt.stopPropagation();
+            evt.preventDefault();
+            const room = selectedRoomValue('selectedroom');
+            const userId = userInputValue('userid');
+            canenrolRoom(
+                room,
+                userId,
+            ).then((canenrol) => {
+                if (canenrol) {
+                    callback(null, room, userId);
+                } else {
+                    callback('Can not enrol to this channel');
+                }
+            }
+            ).catch(e => console.log(e));
+        });
+};
+
+const handleSender = (socket, room, user) => {
+    const _messageFactory = messageFactory(room, user);
+    document.getElementById('sendbutton')
+        .addEventListener('click', (evt) => {
+            evt.stopPropagation();
+            const textElement = document.getElementById('message');
+            const message = _messageFactory.compose(textElement.value)
+            socket.emit('message', message);
+        });
+};
+
+document.addEventListener('DOMContentLoaded', () => {
+    let socket;
+    fetchRooms()
+        .then((rooms) => {
+            populateRooms(rooms, 'rooms');
+            handleSelectedRoom();
+        })
+        .catch(e => console.log(e));
+
+    handleEnrolment((err, room, userId) => {
+        if (!err) {
+            const socketParams = {
+                url: baseUrl,
+                channel: room,
+                options: {
+                    query: `user=${userId}`
+                },
+            };
+            socket = createSocket(socketParams);
```

Comment:
```
const socket = createSocket(socketParams)  ?
```
Not sure if missing something with let socket

Reply:
Related with prior comment.